public/query2.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This script will output a CSV containing the keyfob IDs and names associated with all active memberships
+ * 
+ * Our Raspberry Pi's within the space currently poll this script every few minutes. We wish to move this logic within
+ * Laravel later, but we're saving that work until we can develop a more sensible API and can update the space's access
+ * system (either the existing Pi's, or something else that's easier to manage)
+ */
+
+// Enable access to Composer dependencies (we'll indirectly have phpdotenv through our Laravel)
+require __DIR__ . '/../vendor/autoload.php';
+
+$dotenv = Dotenv::load(__DIR__ . '/../');
+Dotenv::required([
+    'QUERY2_ACCESS_KEY',
+    'DB_HOST',
+    'DB_PORT',
+    'DB_DATABASE',
+    'DB_USERNAME',
+    'DB_PASSWORD'
+]);
+
+if ($_GET['key'] !== $_ENV['QUERY2_ACCESS_KEY']) die();
+
+//Our MySQL connection details.
+$host = $_ENV['DB_HOST'];
+$port = $_ENV['DB_PORT'];
+$user = $_ENV['DB_USERNAME'];
+$password = $_ENV['DB_PASSWORD'];
+$database = $_ENV['DB_DATABASE'];
+
+//Connect to MySQL using PDO.
+$pdo = new PDO("mysql:host=$host;port=$port;dbname=$database", $user, $password);
+
+//Create our SQL query.
+$sql = <<<SQL
+SELECT key_fobs.key_id,
+    users.announce_name
+FROM key_fobs
+    LEFT JOIN users ON users.id = key_fobs.user_id
+WHERE (
+        users.status = 'active'
+        OR users.status = 'leaving'
+    )
+    AND key_fobs.Active = '1'
```

Comment:
Are there any other status' that should have access to the space?

https://github.com/HACManchester/membership-system/blob/e251731e5276eb1cb1f793b618b1cc846d306a7f/app/Entities/User.php#L111-L120

We could do with centralising what states we consider "active", different parts of the code are applying independent "is active?" checks with different sets of states. For example:

https://github.com/HACManchester/membership-system/blob/e251731e5276eb1cb1f793b618b1cc846d306a7f/app/Entities/User.php#L199-L215

It does seem like "honorary" and "payment-warning" are the additional states that are considered active?

```suggestion
WHERE users.status IN(
        'active',
        'leaving',
        'honorary',
        'payment-warning'
    )
    AND key_fobs.Active = '1'
```

Reply:
Indeed, we don't appear to use honorary states and I've not personally seen payment warning happen - I'm not *that* sure the billing side of the system uses it either although I haven't had a proper look into it.

Reviewing how the billing side works will be a big bit of work but will need to be done at some point. Hopefully it can be done in small chunks.

I think starting with a FSM of states and transitions between states is where we should start, then look at how the system currently works, and bring the two together. It seems the billing system sends events out which complicate things a bit. 

It would be good to get this side of the system as dumb as bricks so there are fewer chances for errors.